app/services/qa/linked_data/deep_sort_service.rb

```diff
@@ -0,0 +1,244 @@
+      # * preference for numeric sort (if only one value each and both are integers or a string that can be converted to an integer)
+      # * single value sort (if only one value each and at least one is not an integer)
+      # * multiple values sort (if either has multiple values)
+      # @returns the sorted array
+      # @example returned sorted array
+      #   [
+      #     {:uri=>[#<RDF::URI:0x3fcff54a829c URI:http://id.loc.gov/authorities/names/n201002344>],
+      #      :id=>[#<RDF::Literal:0x3fcff4a367b4("n 201002344")>],
+      #      :label=>[#<RDF::Literal:0x3fcff54a9a98("Cornell, Joseph"@en)>],
+      #      :altlabel=>[],
+      #      :sort=>[#<RDF::Literal:0x3fcff54b4c18("1")>]},
+      #     {:uri=>[#<RDF::URI:0x3fcff54a829c URI:http://id.loc.gov/authorities/names/n2010043281>],
+      #      :id=>[#<RDF::Literal:0x3fcff4a367b4("n 2010043281")>],
+      #      :label=>[#<RDF::Literal:0x3fcff54a9a98("Valli, Sabrina"@en)>],
+      #      :altlabel=>[],
+      #      :sort=>[#<RDF::Literal:0x3fcff54b4c18("2")>]}
+      #   ]
+      def sort
+        @sortable_elements.sort.map(&:element)
+      end
+
+      class DeepSortElement
+        attr_reader :element, :literals, :preferred_language
+        private :preferred_language
+
+        delegate :size, to: :@literals
+
+        def initialize(element, sort_key, preferred_language)
+          element[sort_key] = Qa::LinkedData::LanguageSortService.new(element[sort_key], preferred_language).sort
+          @element = element
+          @literals = element[sort_key]
+          @preferred_language = preferred_language
+          @has_preferred_language = includes_preferred_language?
+          @all_same_language = all_same_language?
+        end
+
+        def <=>(other)
+          return numeric_comparator(other) if integer? && other.integer?
+          return single_value_comparator(other) if single? && other.single?
+          multiple_value_comparator(other)
+        end
+
+        # @returns true if there is a single literal that is an integer or a string that can be converted to an integer; otherwise, false
+        def integer?
+          return false unless single?
+          (/\A[-+]?\d+\z/ === literal.to_s) # rubocop:disable Style/CaseEquality
+        end
+
+        def integer(idx = 0)
+          Integer(literal(idx).to_s)
+        end
+
+        # @returns true if there is only one value; otherwise, false
+        def single?
+          @single ||= literals.size == 1
+        end
+
+        def literal(idx = 0)
+          literals[idx]
+        end
+
+        def downcase_string(idx = 0)
+          to_downcase(literal(idx))
+        end
+
+        def language(idx = 0, list = literals)
+          return list[idx].language if list[idx].respond_to?(:language)
+          nil
+        end
+
+        def includes_preferred_language?
+          return @has_preferred_language if @has_preferred_language.present?
+          filtered = filtered_literals(preferred_language)
```

Comment:
Given the literals are sorted by language you could just test `language == preferred_language` since the first item will either be in the prefered language or it is not in the list at all.

Reply:
Done